libsolidity/codegen/LValue.cpp

```diff
@@ -110,6 +110,8 @@ void MemoryItem::storeValue(Type const& _sourceType, SourceLocation const&, bool
 			solAssert(m_dataType->calldataEncodedSize(false) == 1, "Invalid non-padded type.");
 			if (m_dataType->category() == Type::Category::FixedBytes)
 				m_context << u256(0) << Instruction::BYTE;
+			if (m_dataType->category() == Type::Category::Bool)
```

Comment:
The call to `utils.convertType(_sourceType, *m_dataType, true)` above calls `cleanHigherOrderBits` (because of the `true` flag).